db/pebble/db_test.go

```diff
@@ -391,3 +391,38 @@ func TestNext(t *testing.T) {
 		require.NoError(t, it.Close())
 	})
 }
+
+func TestPanic(t *testing.T) {
+	testDB := pebble.NewMemTest()
+	t.Cleanup(func() {
+		require.NoError(t, testDB.Close())
+	})
+
+	t.Run("view", func(t *testing.T) {
+		defer func() {
+			p := recover()
+			require.NotNil(t, p)
+		}()
+
+		require.NoError(t, testDB.View(func(txn db.Transaction) error {
+			panic("view")
+		}))
+	})
+
+	t.Run("update", func(t *testing.T) {
```

Comment:
I commented locally this line `defer discardTxnOnPanic(txn)` it the test pass successfully.
Is there a way to test added changes? I think we need somehow pass the same txn to testDB.View insdie defer closure 🤔

Reply:
`pebble.Close()` returns an error if you have an open read transaction on the database
but if you have a open write transaction, it doesnt for some reason. if you remove the deferred call from `View()` it should fail.

Let me see if I can better test the change on `Update()` as well.

Reply:
Updated the test.